### ISSUE TITLE
Maintain images original size and quality when resizing them

### ISSUE THREAD

**Issue by casainho** [giuspen/cherrytree#226]
I would prefer that images could maintain images original size and quality when resizing them. Right now, if I resize down an image, it will loose the quality and size of image - this means I can resize down it temporary and later If I try to see the full quality image, it will not be possible, it will be lost :-(

I would like to keep the original image to be able to have it in future - because it is like archiving the image, even if I prefer to see it with a small size inside a text, I will want to see the full details of image when needed.
Maybe this could be as an option when inserting images...

OneNote works like this, keeping the original image, never loosing quality.

Thanks!!

**Comment by philippHRO** [comment]
I approve of this ;)

**Comment by tinki-winki** [comment]
I also support this, it is keeping me from using cherry tree. I will start using the application the moment that the issue is solved.
Thank you!

**Comment by ibrshv** [comment]
I support this.

**Comment by clement-lt** [comment]
I support this. This would be an important feature for my use of cherrytree.

This is how it works in keepnote, and IMHO this is its only advantage over cherrytree.

**Comment by kynanelliot** [comment]
Yes, please I support this too. Only thing holding me back from using CherryTree.

**Comment by philippHRO** [comment]
I support this idea, **but** in my opinion, there will have to be some sort of compression, that is automatically applied *in some cases*.  Because a user might paste a BMP. BMP quickly reach large filesizes, and that can blow up the CherryTree database considerably. In these cases, it would probably be a good idea to at least convert them to PNG.

In the preferences of CT, there could be 2 options:
* 'Automatically convert images to {jpg|png}'
* 'Do not convert images, leave file as is.'

**Comment by dardoor** [comment]
+1 to leaving the same quality/size. I also wish the files would stay in their original format. Currently when you add a JPG it gets converted to PNG, which is very wasteful, since it only makes the files bigger. I'm not sure if this is a bug or not, I'll open an issue just in case.

@kettenfett: BMPs are already converted to PNG in CT. And I agree that's good, because it saves space and there are no downsides.
But any other conversion (like PNG->JPG) is not needed, IMO. If a user adds a PNG file it's probably because he needs it in that format. And if it's PNG it's quite likely from a "computer" source like a screenshot, so saving to JPG might not even make the file smaller. I think it would mean additional UI bloat and programming work for little or no benefit.
